### PR TITLE
Validate credentials when performing a login

### PR DESCRIPTION
**Pull request by tylerslaton**
Adding in a new function that validates the credentials against the specified registry. If the validation fails it will return an error and not add it to the Acorn credentials.

Issue #324

### CODE REVIEW

File: pkg/cli/credential_login.go [hunk 1]
```diff
@@ -89,3 +98,23 @@ func (a *CredentialLogin) Run(cmd *cobra.Command, args []string) error {
 	fmt.Println(cred.Name)
 	return nil
 }
+
+// validateCredentials takes a CredentialLogin and host string to validate
+// whether their combination is valid and will succeed login
+func validateCredentials(ctx context.Context, cred *CredentialLogin, host string) error {
```
Comment (cjellick) [inline pkg/cli/credential_login.go:104]: @ibuildthecloud need your input here. Do you want to do this validation client (cli) side or do you want it done server side?

Comment (tylerslaton) [reply pkg/cli/credential_login.go:104]: If we do decide on this, I have a change locally that I can easily push up to make it be server side.

Comment (vincent99) [reply pkg/cli/credential_login.go:104]: I would think server, an internal registry might be reachable by the cluster but not the client.

Comment (cjellick) [reply pkg/cli/credential_login.go:104]: yeah, server side, go for it

Comment (tylerslaton) [reply pkg/cli/credential_login.go]: Alright, the latest push moved this server side.

**Comment by tylerslaton** [comment]
Working through some issues moving to server side where the existing tests are validating against registries that do not exist.

**Comment by tylerslaton** [comment]
Okay, got the tests passing again. Looks like there was an issue with the listing functionality not properly running.

**Review by cjellick — APPROVED** [review]

**Comment by cjellick** [comment]
@tylerslaton - just an async heads up: this is actually still client side pkg/client is called by pkg/cli, before the call goes to the server. To get this call server side, it should have landed in https://github.com/acorn-io/acorn/blob/main/pkg/server/registry/apps/strategy.go

No action needed on your part, I'm going to see if we can fix today as part of v0.4.1. Just wanted to let you know asyncronously for future reference (assuming you get a notification of this comment :-) )

**Comment by tylerslaton** [comment]
@cjellick Ah, right on. Thanks!

**Pull request merged**